wagtail_newsletter/models.py

```diff
@@ -156,10 +157,14 @@ def get_newsletter_template(self) -> str:
     def get_newsletter_context(self) -> "dict[str, Any]":
         return {"page": self}
 
-    def get_newsletter_html(self) -> SafeString:
+    def get_newsletter_html(self, extra_context=None) -> SafeString:
+        context = self.get_newsletter_context()
+        if extra_context:
+            context = copy(context)
```

Comment:
Question: why copy the context dict? It seems like an unnecessary step given that it isn't reused. If observing immutability of the dict is important for some reason, you could use the [dict union operator](https://peps.python.org/pep-0584/) and skip the import.